packages/core/src/browser/shell/application-shell.ts

```diff
@@ -1618,12 +1618,32 @@ export class ApplicationShell extends Widget {
         return [...this.tracker.widgets];
     }
 
-    canToggleMaximized(): boolean {
-        const area = this.currentWidget && this.getAreaFor(this.currentWidget);
-        return area === 'main' || area === 'bottom';
+    canToggleMaximized(widget: TabBar<Widget>): boolean {
```

Comment:
@Anasshahidd21 you can make `widget` optional as to not be a breaking change (the checks are already performed).

Reply:
done.